Network Class/Network.cpp

```diff
@@ -503,6 +512,106 @@ Network::Network(const string& networkFilename, const string& validationDataFile
 	Classify(validationDataFilename);
 }
 
+//validateInputs() checks all the inputs the user enters or the class gets from the file
+//However, this only checks the hyperparameters for now but I am working on the method to validate the input and output layer size by parsing through the file
+//This method takes two parameters, one is the layersizes string and the other one is range for learning Rate (which is by default set to 0 to 1)
+//It returns a vector of string which contains all the wrong inputs entered by the user or read from the file 
+//That can be used later on to display on the screen while debugging to let the user know what was wrong
+
+//	std::vector<string>	validateInputs(string&, const string& = ("0 1"));   --->prototype in class `Network`
+
+std::vector<string> Network::validateInputs(string& str, const string& lr_range)
+{
+	bool result;
+	int go_ahead(0);
+	string temp;
+	int j(0);
+	//int end_of_file = filesize(trainingDataInfile);
+	std::vector<string> wrong_inputs;
+	std::vector<double> range = Strtok<double>(lr_range, " ");
+	while (!(result = ((learningRate > range[0]) && (learningRate < range[1]))))	//loop continues until the learning rate is between two given end points
```

Comment:
why specify an upper bound for the learning rate? It may be unwise to use 3000 for eta, but that doesn't mean there will never be a scenario when it may be appropriate. As long as it's greater than 0, pretty much any real number will work.

Reply:
Actually I am planning to have a setter for learning Rate. So that the programmer can set a certain limits on the value of the learning rate.

Reply:
Yeah definitely, it can be broken in 4 mutator methods: one for each hyperparameter that user enters or we receive from the file. And also, I was looking forward to have a mutator for setting up any directory for the user. So, user won't have to type in the complete location of the file while giving the training data file name or expected values file name. I can go ahead and code those mutators if you don't mind.

Reply:
If you write it before I get around to it, then by all means.

Reply:
Yeah I just updated the .cpp